pkg/rbac/cache.go

```diff
@@ -4,20 +4,25 @@ package rbac
 import (
 	"sync"
 
+	"github.com/jackc/pgx/v4/pgxpool"
+	// db "github.com/stolostron/search-v2-api/pkg/database"
 	v1 "k8s.io/client-go/kubernetes/typed/authentication/v1"
 )
 
 // Cache used to optimize requests to the Kubernetes API server.
 type Cache struct {
 	authClient          v1.AuthenticationV1Interface // This allows tests to replace with mock client.
+	pool                *pgxpool.Pool
```

Comment:
In the indexer we use `pgxpoolmock.PgxPool`.  [Link to code](https://github.com/stolostron/search-indexer/blob/main/pkg/database/connection.go#L21)

I believe this is because `pgxpoolmock` can accept a `pgxpool`, but the opposite didn't work.